core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -62,7 +62,8 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
 
   private[spark] def loadFromSystemProperties(silent: Boolean): SparkConf = {
     // Load any spark.* system properties
-    for ((key, value) <- Utils.getSystemProperties if key.startsWith("spark.")) {
+    for ((key, value) <- Utils.getSystemProperties if key.startsWith("spark.")
+      || key.startsWith("pyspark.")) {
```

Comment:
I think it's better to live with the slight redundancy instead of adding yet another "blessed prefix". Who knows what other places hardcode this restriction (e.g. `Utils.loadDefaultSparkProperties` would fail to ready `pyspark` properties).